azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+        sample_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        to_set_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        to_set_kv.value = to_set_kv.value + "a"
+        to_set_kv.tags = {"a": "b", "c": "d"}
+        to_set_kv.etag = "wrong etag"
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+        self.app_config_client.delete_configuration_setting(
+            to_set_kv.key, label=to_set_kv.label
+        )
+
+    def test_set_configuration_setting_no_label_etag(self):
+        to_set_kv = copy(self.test_data.no_label_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+
+    def test_set_configuration_setting_label_etag(self):
+        to_set_kv = copy(self.test_data.label1_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+
+    def test_set_configuration_setting_no_label_no_etag(self):
+        to_set_kv = copy(self.test_data.no_label_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        to_set_kv.etag = None
+        set_kv = self.app_config_client.set_configuration_setting(to_set_kv)
+        # remove immediately from the AzConfig service
+        self.app_config_client.delete_configuration_setting(set_kv.key)
+        assert (
+            to_set_kv.key == set_kv.key
+            and None is set_kv.label
+            and to_set_kv.value == set_kv.value
+            and to_set_kv.content_type == set_kv.content_type
+            and to_set_kv.tags == set_kv.tags
+            and to_set_kv.etag != set_kv.etag
+        )
+
+    def test_set_configuration_setting_label_no_etag(self):
+        to_set_kv = copy(self.test_data.label1_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        to_set_kv.etag = None
+        set_kv = self.app_config_client.set_configuration_setting(to_set_kv)
+        # remove immediately from the AzConfig service
+        self.app_config_client.delete_configuration_setting(
+            set_kv.key, label=set_kv.label
+        )
+        assert (
+            to_set_kv.key == set_kv.key
+            and to_set_kv.label == set_kv.label
+            and to_set_kv.value == set_kv.value
+            and to_set_kv.content_type == set_kv.content_type
+            and to_set_kv.tags == set_kv.tags
+            and to_set_kv.etag != set_kv.etag
+        )
+
+    # method: update_configuration_setting
+    def test_update_existing_configuration_setting_no_label_etag(self):
+        sample_kv = copy(self.test_data.no_label_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_" + self.test_data.key_uuid
```

Comment:
what happens if you run the tests in parallel? it looks like you are using the same key for each test. if 2 run at the same time and one is updating and the other deleting, there will be problems